source/MaterialXContrib/MaterialXNode/MaterialXData.h

```diff
@@ -3,16 +3,18 @@
 
 #include <MaterialXCore/Document.h>
 #include <MaterialXGenShader/GenContext.h>
-
-#include "../OGSXMLFragmentWrapper.h"
+#include <MaterialXContrib/OGSXMLFragmentWrapper.h>
 
 #include <maya/MString.h>
 
 #include <vector>
 
+using OGSXMLFragmentWrapperPtr = std::unique_ptr<MaterialX::OGSXMLFragmentWrapper>;
```

Comment:
I'm not a fan of typedefs like this TBH. The reader is left wondering what kind of smart pointer it is. MaterialX seems to be using shared_ptr everywhere but this is a unique_ptr which has completely different semantics and yet the typedef looks the same. 

Besides, it's not like there's just one possible smart pointer type for OGSXMLFragmentWrapper. The choice of the smart pointer depends on particular owner, not the owned class.

Reply:
OGSXMLFragmentWrapperUniquePtr ?
We generally get rid of implementation details from the interface.

Reply:
UniquePtr is definitely better, maybe UPtr could work for brevity.

Well, a smart pointer is not an interface, users have to be aware of the exact semantics.

Reply:
I got rid of the exposure here as it's not required so moot point for naming :)